src/client.ts

```diff
@@ -1105,16 +1108,17 @@ export class StreamChat<
           throw new Error('Invalid request type');
       }
       this._logApiResponse<T>(type, url, response);
-
+      this.consecutiveFailures = 0;
       return this.handleResponse(response);
     } catch (e) {
       this._logApiError(type, url, e);
-
+      this.consecutiveFailures += 1;
       if (e.response) {
         if (
           e.response.data.code === chatCodes.TOKEN_EXPIRED &&
           !this.tokenManager.isStatic()
         ) {
+          await sleep(this._retryInterval());
```

Comment:
I think we can retry immediately one time. In the best case scenario token gets refreshed from the first attempt

Reply:
sure, modifying this.